### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by 0xTim**
Make it easier to find the section about different logging backends. It's currently buried in a level 4 heading which doesn't make much sense. Also tweak the language as it's a reasonably mature ecosystem now

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,9 +60,9 @@ LoggingSystem.bootstrap(StreamLogHandler.standardError)
 For further information, please check the [API documentation][api-docs].
 
 <a name="backends"></a>
-#### Selecting a logging backend implementation (applications only)
```
Comment (tomerd) [inline README.md:63]: the (applications only) part is important so we should find a place to say that

**Comment by tomerd** [comment]
@swift-server-bot add to allowlist

File: README.md (hunk 1 above)
Comment (0xTim) [reply README.md:63]: Updated

**Comment by tomerd** [comment]
fixing CI in https://github.com/apple/swift-log/pull/281

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Review by ktoso — APPROVED** [review]
Seems we forgot tom merge here, looks good LGTM

**Pull request merged**